### DISCUSSION TITLE
Why is the connection much slower for versions 1.2.0 and 1.2.1 compared to 1.1.9?

### DISCUSSION THREAD

**Discussion by zhangbo8418** [rustdesk/rustdesk#5002] [General]
### Bug Description

 Why is the connection much slower for versions 1.20 and 1.21 compared to 1.19? Connecting for the first time on 1.2x takes 15-30 seconds, while on 1.19 it only takes 3 seconds...

### How to Reproduce

 Why is the connection much slower for versions 1.20 and 1.21 compared to 1.19? Connecting for the first time on 1.2x takes 15-30 seconds, while on 1.19 it only takes 3 seconds...

### Expected Behavior

 Why is the connection much slower for versions 1.20 and 1.21 compared to 1.19? Connecting for the first time on 1.2x takes 15-30 seconds, while on 1.19 it only takes 3 seconds...

### Operating system(s) on local side and remote side

MacOS Ventura1 3.2.1

### RustDesk Version(s) on local side and remote side

1.2x

### Screenshots

<img width="1132" alt="QQ20230714-153030@2x" src="https://github.com/rustdesk/rustdesk/assets/46797568/5f425f23-ddd3-4a5c-988e-2e81ae4a8376">

**Comment by Danxity** [comment]
I am having the same issue. But I'm using Windows 10. I use to use Anydesk and switched over to RustDesk, but now RustDesk seems un-useable. Its so laggy I can't really do anything anymore. And I don't like how new RustDesk don't allow separate windows. I need to control more than 1 pc simultaneously.

Also i tried changing the connection to reaction time over quality, and this function doesn't seem to work. It doesn't apply it seems. The more you update the rustdesk it seems the more problems it starting to have. you guys need to do hard testing before releasing any new versions in my opinion. The older rustdesk which didn't have dark mode yet was way better software. I love the new dark mode, but everything doesn't work good anymore after you presented the dark mode feature.

I tried going back to Anydesk and I find Anydesk is running much smoother than RustDesk at the moment now. 

Why is each update getting worse?

**Reply by kozman1000** [reply]
It could be because more and more people are discovering it day by day and the few public relay servers are probably being saturated as a result. Could also be AV1? There was quite a lapse of time between 1.1.9 and 1.2.x so probably bugs crept in that are still not yet resolved or apparent. You're not paying for it so why complain about a free solution?

**Reply by rustdesk** [reply]
As for quality adjustment, we just fixed yesterday (was broken in one commit), please try out latest nightly build.
As for seperate window, please check https://github.com/rustdesk/rustdesk/discussions/2889, @fufesou is still working on it.

**Comment by kozman1000** [comment]
Definitely not seeing a slower connection when coming into a 1.1.9 or 1.2.x endpoint from my 1.2.1 PC. It's actually more robust and snappier. Might be a Mac specific issue? Have you tried bumping to Ventura 13.3.x/13.4.1?

**Comment by lamithe** [comment]
my connection time is normal. but the control is too laggy(only 5fps on H264/265 with hardware encode).

**Reply by kozman1000** [reply]
You don't have Default Codec set to Auto and Default Image Quality set to Balanced? What hardware are you running this on? I'm in an environment of mainly laptop endpoints that are from circa 2014-2018 running Win 10 x64 with 4GB to 8GB RAM on average and am not hitting this slowness with default 1.1.9 or 1.2.x settings. Are your video drivers current?

**Reply by lamithe** [reply]
i am using a desktop runing Win11 x64 with 32GBRAM, RTX3060 withe newest driver and i7 8700 CPU. every default setup has been tried, but the FPS is still low. with compare, the toDesk can be 30FPS.

**Reply by 21pages** [reply]
https://github.com/rustdesk/rustdesk/discussions/3828#discussioncomment-6445481

**Reply by lamithe** [reply]
@21pages yeah, i know this, and i am very looking forward to this

**Reply by 21pages** [reply]
Windows will be ready soon.

**Reply by lamithe** [reply]
@kozman1000 maybe because my remote resolution is 4k, so the data to be encoded and transfered is too big. the author has replied me that they are rewritting the new implement for copying data between CPU and GPU to solve my problem. i am looking forward to that.

**Reply by lamithe** [reply]
@21pages NICE! VERY NICE!!!

**Comment by rustdesk** [comment]
@21pages work with @zhangbo8418 to check why the connection establishment is slower.

**Reply by 21pages** [reply]
@zhangbo8418  Please contact me with cGFnZXMyMUAxNjMuY29t

**Reply by zhangbo8418** [reply]
It seems that the first connection is very slow, but the second time is fine, maybe because I used 1.2.x for the first connection, and then changed to 1.1.9 for the second connection, so I thought it was a problem with this version. . .

**Reply by 21pages** [reply]
Can it be reproduced?

**Reply by rustdesk** [reply]
It makes sense then.

**Reply by rustdesk** [reply]
The first time it alaways tries to punch hole for direct connection, next time, will not. You can use `<id>/r` to force using relay with out direct conneciton attempt. Yes, using relay while direct connection is clever, but this requires a lot of code change, we won't do it in short term.

**Reply by wes1997** [reply]
The same self-built server, version 1.2.1 is several times slower than version 1.1.9. Version 1.2.1 often takes more than 10 seconds and fails to connect successfully, while version 1.1.9 connects successfully in just a few seconds.
I have tried version 1.2.2, but the problem of slow connection still exists. Even when using the forced relay mode to establish a connection, it is still slow compared to version 1.1.9.

**Reply by wes1997** [reply]
1.1.9 quickly establishes a connection with 1.2.1, while the connection establishment between 1.2.1 and 1.1.9 is slow.

**Reply by rustdesk** [reply]
> @zhangbo8418 Please contact me with cGFnZXMyMUAxNjMuY29t

@wes1997 if you can let our guys test your connection for confirming the slowness, please contact above.

**Reply by zhangbo8418** [reply]
LAN is fast.

https://github.com/rustdesk/rustdesk/assets/46797568/b1aa047f-0801-4ca0-b91a-0a8ee3c6f35d

**Reply by zhangbo8418** [reply]
It's just slow on the first connection. Watch the video, it's fine the second time around.

https://github.com/rustdesk/rustdesk/assets/46797568/b556170a-7dcf-480e-ab7f-492476e43c8a

**Reply by zhangbo8418** [reply]
The Pro version server is different from the open source version server.

**Reply by rustdesk** [reply]
Anyway, you'd better use official one, no guarantee for the others.
